core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -1751,17 +1752,17 @@ object GroupCoordinator {
     GroupCoordinator(config, replicaManager, heartbeatPurgatory, rebalancePurgatory, time, metrics)
   }
 
-  private[group] def offsetConfig(config: KafkaConfig) = OffsetConfig(
-    maxMetadataSize = config.offsetMetadataMaxSize,
-    loadBufferSize = config.offsetsLoadBufferSize,
-    offsetsRetentionMs = config.offsetsRetentionMinutes * 60L * 1000L,
-    offsetsRetentionCheckIntervalMs = config.offsetsRetentionCheckIntervalMs,
-    offsetsTopicNumPartitions = config.offsetsTopicPartitions,
-    offsetsTopicSegmentBytes = config.offsetsTopicSegmentBytes,
-    offsetsTopicReplicationFactor = config.offsetsTopicReplicationFactor,
-    offsetsTopicCompressionType = config.offsetsTopicCompressionType,
-    offsetCommitTimeoutMs = config.offsetCommitTimeoutMs,
-    offsetCommitRequiredAcks = config.offsetCommitRequiredAcks
+  private[group] def offsetConfig(config: KafkaConfig) = new OffsetConfig(
```

Comment:
`OffsetConfig` seems not related to `Defaults`.
Shoule we move it to another PR?

Reply:
Defaults depends on `OffsetConfig` object companion which if we remove it from the `OffsetConfig` file will one have `OffsetConfig` class signature.  I vote to keep it here as it is very small change

Reply:
Thanks. Now I see it.

I prefer to have things as atomic as possible so I prefer to move `OffsetConfig` and other `Defaults` dependencies as separate changes prior to this PR. But it's up to you to decide.